source/agora/node/QRCodeInterface.d

```diff
@@ -0,0 +1,278 @@
+/*******************************************************************************
+
+    Provides QR service interface used by the node's web-based management
+    interface
+
+    Using the web interface. This interface is provided by Vibe.d and
+    may be integrated into the node's web-based management in the future.
+
+    Copyright:
+        Copyright (c) 2019-2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.node.QRCodeInterface;
+
+import agora.common.crypto.Key;
+import agora.common.Hash;
+import agora.common.Types;
+import agora.consensus.data.PreImageInfo;
+
+import agora.utils.Log;
+import agora.node.data.LoginInfo;
+import agora.node.data.EncryptionKey;
+import agora.network.Clock;
+
+import vibe.core.core;
+import vibe.data.json;
+import vibe.http.router;
+import vibe.http.server;
+
+import std.datetime.date : DateTime;
+import std.datetime.systime : SysTime, Clock;
+import std.datetime.timezone : UTC;
+
+import core.time : days;
+
+import barcode;
+
+mixin AddLogger!();
+
+/*******************************************************************************
+
+    Class for Agora User's credentials QR services Interface
+
+    A class to provide a login or encryption key to QRs to verify that
+    the owner of the validator from outside agora.
+
+    This interface listens on 127.0.0.1:2829.
```

Comment:
This has to be configurable, otherwise you can't run 2 servers on the same machine (which we do)

Reply:
I will to set this as a configuration option.
Should I run two servers(agora, this) on the same machine in different ports?

Reply:
@Geod24 I will to set this as a configuration option.
Should I run two servers(agora, this) on the same machine in different ports?

Reply:
It should be the same process listening on two ports, yes

Reply:
https://github.com/bpfkorea/agora/blob/1e6bafc901588eef708d8177537a9c3639d777c9/source/agora/common/Config.d#L186-L198
@Geod24 How about using the same port and server as admin that will be applied in the future?

Reply:
@Geod24 ping
Please review the newly changed code.

Reply:
Actually, same address and same port makes sense.

Reply:
If the port configuration is the same as admin, the QR interface is:
Should I consider it with the admin server?

Reply:
Yes

Reply:
I created the admin interface by referring to # 483.
This is the admin interface with different settings than node initialization logic.
In the future, the management or monitoring interface can be added to the admin interface.